src/cache/cache_las.c

```diff
@@ -627,76 +727,41 @@ __wt_las_cursor_position(WT_CURSOR *cursor, uint32_t btree_id, uint64_t pageid)
  *	Remove all records for a given page from the lookaside store.
  */
 int
-__wt_las_remove_block(WT_SESSION_IMPL *session,
-    WT_CURSOR *cursor, uint32_t btree_id, uint64_t pageid)
+__wt_las_remove_block(
+    WT_SESSION_IMPL *session, uint32_t btree_id, uint64_t pageid)
 {
 	WT_CONNECTION_IMPL *conn;
+	WT_CURSOR *cursor;
 	WT_DECL_RET;
-	WT_ITEM las_key;
 	WT_SESSION_IMPL *las_session;
-	uint64_t las_counter, las_pageid, remove_cnt;
-	uint32_t las_id, session_flags;
-	bool local_cursor, local_txn, locked;
+	WT_TXN_ISOLATION saved_isolation;
+	uint64_t decrement_cnt;
+	uint32_t session_flags;
 
 	conn = S2C(session);
-	remove_cnt = 0;
 	session_flags = 0;		/* [-Wconditional-uninitialized] */
 
-	local_cursor = local_txn = locked = false;
-	if (cursor == NULL) {
-		__wt_las_cursor(session, &cursor, &session_flags);
-		local_cursor = true;
-	}
-	las_session = (WT_SESSION_IMPL *)cursor->session;
-
-	/*
-	 * Wrap all of the removes in a transaction, unless this remove is part
-	 * of a larger operation.
-	 */
-	if (local_cursor) {
-		WT_ERR(__wt_txn_begin(las_session, NULL));
-		las_session->txn.isolation = WT_ISO_READ_UNCOMMITTED;
-		local_txn = true;
-	}
+	__wt_las_cursor(session, &cursor, &session_flags);
 
-	/*
-	 * Search for the block's unique prefix and step through all matching
-	 * records, removing them.
-	 */
-	ret = __wt_las_cursor_position(cursor, btree_id, pageid);
-	__wt_writelock(session, &conn->cache->las_sweepwalk_lock);
-	locked = true;
-	for (; ret == 0; ret = cursor->next(cursor)) {
-		WT_ERR(cursor->get_key(cursor,
-		    &las_pageid, &las_id, &las_counter, &las_key));
+	las_session = (WT_SESSION_IMPL *)cursor->session;
+	__las_set_read_uncommitted(las_session, &saved_isolation);
 
-		/*
-		 * Confirm the search using the unique prefix; if not a match,
-		 * we're done searching for records for this page.
-		 */
-		if (las_pageid != pageid || las_id != btree_id)
-			break;
+	WT_ERR(__wt_txn_begin(las_session, NULL));
 
-		WT_ERR(cursor->remove(cursor));
-		++remove_cnt;
-	}
-	WT_ERR_NOTFOUND_OK(ret);
+	ret = __las_remove_block(
```

Comment:
I think a comment here (or at the top) about "We are really a wrapper function setting up the cursor, transaction and needed isolation level. The worker function will acquire needed locks to remove the identified entries." (I don't like that wording exactly, but it's the idea.)

Reply:
I went with:
```
This is an external API for removing records from the lookaside
table, first acquiring a lookaside table cursor and enclosing
transaction, then calling an underlying function to do the work.
```